src/AppBundle/v2/Assembler/ClientAssembler.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace AppBundle\v2\Assembler;
+
+use AppBundle\v2\DTO\ClientDto;
+
+class ClientAssembler
+{
+    /**
+     * @param array $data
+     * @return ClientDto
+     */
+    public function assembleFromArray(array $data)
+    {
+        $this->throwExceptionIfMissingRequiredData($data);
```

Comment:
Maybe 
if (!$this->dataIsValid($data)) {           
 throw new \InvalidArgumentException(__CLASS__ . ': Missing all data required to build DTO');
}
or even rename dataIsValid() to validateData() and put it in an interface? to ensure all future assemblers have that method?

Reply:
Using the software rule of 3, I'd rather not break anything out into abstract classes or interface methods yet. Your top suggestion is pretty much what is happening inside that method.